First draft of how to generate it from bibtex

smarr opened the pull request:
Just a bit playing around.

I thought before I invest time into adding more material, I may as well try to get to benefit from existing bibtex entries.

smarr committed:
First draft of how to generate it from bibtex

Signed-off-by: Stefan Marr <git@stefan-marr.de>

smarr committed:
Fixed edge for edges without labels

Signed-off-by: Stefan Marr <git@stefan-marr.de>

smarr committed:
Fixed dependencies order

Signed-off-by: Stefan Marr <git@stefan-marr.de>

smarr committed:
Added more stuff

Signed-off-by: Stefan Marr <git@stefan-marr.de>

eregon commented:
Looks interesting :)
That's a lot of edges though, e.g. for Lookup Caches. I would simplify so papers extending the first one implicitly are in the same concept.
Separating extends/uses seems a good idea :+1:

eregon commented on bibtex.html:
This should be a `uses` edge

smarr commented:
Yeah, I thought I add a couple of things before refining.
I'll try to add the other entries and then we can perhaps have an informed discussion.

eregon commented on bibtex.html:
I think this edge should be an `extends` of the Domani reachability paper.
Something like Domani uses the concept `Tracking Reachability to Reduce Synchronization` and this paper extends Domani.
I didn't know about the Domani paper when writing this paper, but the technique is fairly similar so it could be seen as an extension of Domani. It also extends what's done with reachability, so I'd say `extends` over `uses` here.

`extends`/`uses` edges won't always be based on the author knowing about the previous paper beforehand, but I think that's fine.

smarr commented:
I'll process the edge details later, I think. Just acknowledging that I am seeing it :)
Though, the bib stuff really needs to go into a separate file first (it already is, just sharing was easier with embedded in html).

smarr committed:
More stuff

Signed-off-by: Stefan Marr <git@stefan-marr.de>

smarr committed:
Refactor code and add support for edges between concepts

Signed-off-by: Stefan Marr <git@stefan-marr.de>

smarr committed:
Added more stuff and comments

Signed-off-by: Stefan Marr <git@stefan-marr.de>

smarr committed:
Escape ampersands in URLs otherwise rendering breaks

Signed-off-by: Stefan Marr <git@stefan-marr.de>

smarr committed:
Avoid redundant edges to concepts

Signed-off-by: Stefan Marr <git@stefan-marr.de>

smarr committed:
Replace .dot with .bib file

Signed-off-by: Stefan Marr <git@stefan-marr.de>

smarr committed:
Show details on mouse over

Signed-off-by: Stefan Marr <git@stefan-marr.de>

smarr commented:
I haven't yet addressed the comments, but I added a few more things, and am at a point where I probably should ask what you think before adding more stuff :)

I did a bit out automated polishing to remove redundant edges to concepts, but there's probably more that can be done.
Generally, I'd want the data to be as complete as possible, and then simply filtering out stuff.

I also added a mouse over to show the details of the element, though, this needs some visual polish.

smarr committed:
Set font to something modern sans-serif

Signed-off-by: Stefan Marr <git@stefan-marr.de>

smarr committed:
Restrict the height and width so that initial view is zoomed

Signed-off-by: Stefan Marr <git@stefan-marr.de>

smarr committed:
Render bibtex and link to github

Signed-off-by: Stefan Marr <git@stefan-marr.de>

eregon commented on bibtex.html:
```diff
@@ -0,0 +1,391 @@
+// const dataPromise = fetch('https://raw.githubusercontent.com/eregon/virtual-machines-research-graph/master/vm.bib')
+//   .then((response) => {
+//     return response.text();
+//   });
```
Should we remove this commented code, or rather replace the one below with it?

eregon reviewed (approved):
It looks great! Let's merge it and tweak it further after.

eregon commented on bibtex.html:
```diff
@@ -0,0 +1,391 @@
+<!DOCTYPE html>
```
Probably we should remove the old `index.html` and replace it by this, I can do that after merging.

eregon commented on vm.bib:
Typo here

eregon commented on vm.bib:
This extends should be for SELF Maps, not TOSM.

smarr replied:
Fixed

smarr committed:
Determine URL based on whether we run on github or not

Signed-off-by: Stefan Marr <git@stefan-marr.de>

smarr committed:
Process comments on content

Signed-off-by: Stefan Marr <git@stefan-marr.de>

smarr committed:
Delete index.html

Signed-off-by: Stefan Marr <git@stefan-marr.de>

smarr committed:
Rename bibtex.html to index.html

Signed-off-by: Stefan Marr <git@stefan-marr.de>

smarr committed:
More content fixes

Signed-off-by: Stefan Marr <git@stefan-marr.de>

smarr commented:
Think that last push fixed the remaining bits.
